### ISSUE TITLE
test_sort failures with num_warps=8 after agama driver upgrade

### ISSUE THREAD

**Issue by whitneywhtsang** [intel/intel-xpu-backend-for-triton#6869]
## Description

`test_sort` with `num_warps=8` produces incorrect results across all tested dtypes (float16, float32, bfloat16, int32). The sort output contains wrong elements (not just wrong ordering), suggesting a code generation issue in the multi-warp sort kernel.

## Affected tests

All `test_sort` parametrizations with `num_warps=8` and multiple rows/columns:
- `test_sort[float16-{True,False}-8-{1-512,8-64,256-16}]`
- `test_sort[float32-{True,False}-8-{1-512,8-64,256-16}]`
- `test_sort[bfloat16-{True,False}-8-{1-512,8-64,256-16}]`
- `test_sort[int32-{True,False}-8-{8-64,256-16}]`

## Root cause

The failures appear after the **icpx compiler upgrade from 2025.3.2 to 2025.3.3**:
- Passing CI run (icpx 2025.3.2): https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/25470001539/job/74743210728
- Failing CI run (icpx 2025.3.3): https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/25475722546/job/74828931260

Same triton commit, same PyTorch commit (`34d845da`), different compiler version on the runners.

## Reproduction

```
python -m pytest python/test/unit/language/test_standard.py::test_sort -k "8-8-64" --device xpu
```

Requires icpx 2025.3.3 environment.

## Error pattern

The sort results contain incorrect values — not just misordered, but entirely wrong elements appear in the output, suggesting the multi-warp shuffle/exchange in the sort kernel is being miscompiled.

Example:
```
Expected: tensor([[-3.3999, -3.2521, -2.9601, -2.7592, -2.4201, -2.4132, -2.2161, -2.1439]])
Got:      tensor([[-3.3999, -2.9601, -2.7592, -2.4201, -2.4132, -2.2161, -2.0946, -2.0360]])
```

**Comment by anmyachev** [comment]
@wdziurdz the reason should be the same as for inductor topk problems

**Comment by wdziurdz** [comment]
Root cause: the bug isn't in Triton.  tl.topk returns the correct values  was verified by comparing the kernel output against a CPU reference (instead of torch.topk xpu ) and it matches. The test in test_standard.py  uses torch.topk(x_xpu) as its reference, and that's what's actually broken.

The regression was introduced  in  torch-xpu-ops PR [#3371](https://github.com/intel/torch-xpu-ops/pull/3371) which added a "subgroup topk" SYCL kernel. It was reverted in [#3707](https://github.com/intel/torch-xpu-ops/pull/3707) on 2026-05-19, but our pinned pytorch older.

Repro without Triton on the currently pinned torch:
```python
import torch
torch.manual_seed(0)
a = torch.randint(0, 999, [1,1,8,8], dtype=torch.float32)
cpu = torch.topk(a, 2, -1).values
xpu = torch.topk(a.xpu(), 2, -1).values.cpu()
print(torch.allclose(cpu, xpu))
```

No issue on this version 3.13.0.dev20260520+xpu from nightly https://download.pytorch.org/whl/nightly/torch/   with this revert [#3707](https://github.com/intel/torch-xpu-ops/pull/3707) 

**Next steps:** waiting on new PyTorch pin .

**Comment by wdziurdz** [comment]
The Triton kernel itself is correct  when the reference values are computed on CPU instead of XPU, the Triton output matches exactly. So the mismatch lives entirely in torch.topk on XPU.

 Standalone repro (no Triton):
```python
  import torch
  torch.manual_seed(0)
  a = torch.randint(0, 999, [1, 1, 8, 8], dtype=torch.float32)
  print(torch.allclose(torch.topk(a, 2, -1).values,
                       torch.topk(a.xpu(), 2, -1).values.cpu()))
```

The updated version of PyTorch did not help. The potential fix doesn't  work, and an issue has been created in torch-xpu-ops: https://github.com/intel/torch-xpu-ops/issues/3777

**Comment by wdziurdz** [comment]
Also need revert this commit : https://github.com/intel/intel-xpu-backend-for-triton/pull/7562/changes/6de7e868f74711cf820cded172c7781f5381e987

**Comment by anmyachev** [comment]
Thanks @wdziurdz!
